functorch/_src/aot_autograd.py

```diff
@@ -948,26 +987,24 @@ def aot_function_simplified(
             num_params_buffers=params_len,
         )
 
-        compiled_fn = None
+        def compile(fn, *args):
+            return create_aot_dispatcher_function(
+                fn,
+                args,
+                aot_config,
+                fake_mode,
+            )
+
+        compiled_fn = compile(fn, *fake_flat_tensor_args, *inputs)
 
         @wraps(fn)
         def new_func(*args):
-            nonlocal compiled_fn
-            # Disabling the cache because it causes some models to fail
-            if True:
-                compiled_fn = create_aot_dispatcher_function(
-                    fn,
-                    args,
-                    aot_config,
-                )
             return compiled_fn(args)
-
         return new_func
 
     compiled_f = aot_function_simplified(functional_call, *top_args, **top_kwargs)
 
     if top_kwargs:
-
         def forward(*args, **kwargs):
             return compiled_f(
                 *params_flat,
```

Comment:
Aren't these still non fake? What's going on here?

Reply:
Yes, these should be called with real args, because this is real "forward" so the params must be the real ones from the model. I know its confusing :(

Reply:
Added explanation in the pr description.